runtime/gc_modron_startup/mminit.cpp

```diff
@@ -3196,7 +3199,78 @@ gcReinitializeDefaultsForRestore(J9VMThread* vmThread)
 		extensions->gcThreadCountForced = false;
 	}
 
-	return result;
+	/* While ideally it is desired to change the heap geometry (parameters such as
+	 * extensions->maxMemory, virtual memory reserved, etc), it would be too complex
+	 * and slow. As an alternative, the geometry remains unchanged, but the
+	 * existing softmx functionality is utilized to limit the maximum size of heap.
+	 */
+
+	{
+		uintptr_t candidateSoftMx = 0;
```

Comment:
put a comment  like 'initially set candidate softMx as a function of usable physical memory, and then further/later adjust or refuse based on Xmx/Xms values and other conditions'